### PR TITLE
wasm: fix "else-conflict 2"

### PR DESCRIPTION
**Pull request by srenatus**
There are two kinds of else conflicts in the e2e tests, and one had been failing in WASM (i.e., it didn't fail when it should).

With this change to the planner, the underlying issue is resolved:

With multiple rule bodies for one rule like this, where r1 and r2 both fail,

- r1
  - else1
- r2
  - else2

previously, else1 and else2 checked if the output variable used for both r1 and r2 was defined, and didn't run it was.
Topdown, however, evaluates both else1 and else2, and checks that they have the same outcome.

The output variable checking in WASM would prohibit else2 from being run when the output variable was set by else1.

**Now**, each rule and its else branches use an output variable _different from the overall output_; and an extra block writing that output variable's value into the overall output variable of that rule. The extra layering allows each else block to check **its rule body's output**, and not the overall rule output.

### CODE REVIEW

File: internal/wasm/sdk/test/e2e/external_test.go [hunk 1]
```diff
@@ -121,19 +121,13 @@ func assert(t *testing.T, tc cases.TestCase, result *opa.Result, err error) {
...
-		if tc.WantErrorCode != nil {
-			t.Logf("\nExpected Code: %s\nGot Err: %s\n", *tc.WantErrorCode, err)
-		}
-
-		if tc.WantError != nil {
-			t.Logf("\nExpected Err: %s\nGot Err: %s\n", *tc.WantError, err)
-		}
-
-		t.Skip("Skipping test case: Error validation not supported")
+		assertErrorCode(t, *tc.WantErrorCode, err)
```
Comment (srenatus) [inline internal/wasm/sdk/test/e2e/external_test.go:130]: This part of the test change kind of assumes that #3022 is done. 🙈

File: internal/planner/planner_test.go [hunk 2]
```diff
@@ -224,6 +224,42 @@ func TestPlannerHelloWorld(t *testing.T) {
...
+
+				q {
+					false
+				}
+				else = true {
+					true
+				}
+				q = false
+				`,
+			},
```
Comment (srenatus) [inline internal/planner/planner_test.go:241]: ### Before

I don't think what we had before was wrong, but this is how it changed:
```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 3 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (2 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:3 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:2 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | *ir.Block Block (5 statements)
| | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

### After

```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 4 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = true}}
| | | *ir.Block Block (5 statements)
| | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:9 Row:9 file:module-0.rego text:false}}
| | | | *ir.MakeBooleanStmt &{Value:false Target:7 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarStmt &{Source:7 Target:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (2 statements)
| | | | *ir.IsDefinedStmt &{Source:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | | *ir.AssignVarOnceStmt &{Target:2 Source:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

File: internal/planner/planner_test.go [hunk 3]
```diff
@@ -224,6 +224,42 @@ func TestPlannerHelloWorld(t *testing.T) {
...
+				else = false {
+					true
+				}
+				q {
+					false
+				}
+				else = true {
+					true
+				}`,
+			},
```
Comment (srenatus) [inline internal/planner/planner_test.go:261]: Here, what we had before was indeed wrong:

### Before

```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 3 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (2 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:3 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:2 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (2 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:3 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:2 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:3 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:3 B:4 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:5 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

The second `else` block's `IsUndefinedStmt` checks the `fn.Return` local:
```
| | | | | | *ir.IsUndefinedStmt &{Source:2 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
```
...which is **not undefined** after the first `else` block has been executed.


### After

```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 3 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:7 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (5 statements)
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:4 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:4 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:5 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:5 B:6 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:7 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarStmt &{Source:7 Target:4 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:4 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:4 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

Now, we've got one local per set of "rule+elses", and each `else` only checks if the corresponding rule's local is undefined,

```
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
```
for the first else block (return local 3) and
```
| | | | | | *ir.IsUndefinedStmt &{Source:4 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
```
for the second (return local 4).

Each rule+elses set gets one block for the rule, one per each, and one block for AssignVarOnceStmt, guarded by IsDefinedStmt.

**Review by tsandall — COMMENTED** [review]
LGTM. I wonder about the overhead of allocating an extra variable per block (e.g., suppose there there are 10,000 rules in the set as a result of PE) but it's probably not an issue at this time.

**Comment by srenatus** [comment]
> I wonder about the overhead of allocating an extra variable per block (e.g., suppose there there are 10,000 rules in the set as a result of PE) but it's probably not an issue at this time.

Alternatively, we could introduce a "reset-local" IR stmt that resolves into `i32.const 0; local.set $x`; and re-use the same variable. It would use mean some more instructions, but less locals. 🤔

File: internal/planner/planner_test.go (hunk 3 above)
Comment (srenatus) [reply internal/planner/planner_test.go:261]: With the re-used local, we now have:

```
| *ir.Funcs Funcs (1 funcs)
| | *ir.Func g0.data.p.q (2 params: [0 1], 3 blocks)
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.ResetLocalStmt &{Target:3 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:4 B:5 Location:{Index:0 Col:6 Row:4 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:6 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:6 Target:3 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:4 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:4 B:5 Location:{Index:0 Col:6 Row:7 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:6 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.AssignVarStmt &{Source:6 Target:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:3 Location:{Index:0 Col:5 Row:6 file:module-0.rego text:else = false}}
| | | *ir.Block Block (1 statements)
| | | | *ir.BlockStmt BlockStmt (3 blocks) &{0 0 0  }
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.ResetLocalStmt &{Target:3 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:4 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.NotEqualStmt &{A:4 B:5 Location:{Index:0 Col:6 Row:10 file:module-0.rego text:false}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:6 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | | *ir.AssignVarStmt &{Source:6 Target:3 Location:{Index:0 Col:5 Row:9 file:module-0.rego text:q}}
| | | | | *ir.Block Block (6 statements)
| | | | | | *ir.IsUndefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:4 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:false Target:5 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.NotEqualStmt &{A:4 B:5 Location:{Index:0 Col:6 Row:13 file:module-0.rego text:true}}
| | | | | | *ir.MakeBooleanStmt &{Value:true Target:6 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarStmt &{Source:6 Target:3 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | *ir.Block Block (2 statements)
| | | | | | *ir.IsDefinedStmt &{Source:3 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | | | | *ir.AssignVarOnceStmt &{Target:2 Source:3 Location:{Index:0 Col:5 Row:12 file:module-0.rego text:else = true}}
| | | *ir.Block Block (1 statements)
| | | | *ir.ReturnLocalStmt &{Source:2 Location:{Index:0 Col:5 Row:3 file:module-0.rego text:q}}
```

Note that it's resetting the local `3` in each block.

**Review by tsandall — DISMISSED** [review]
LGTM

File: internal/wasm/sdk/test/e2e/external_test.go [hunk 4]
```diff
@@ -204,7 +198,26 @@ func assertResultSet(t *testing.T, want []map[string]interface{}, sortBindings b
...
+		found := false
+		for _, exp := range exps {
+			if strings.Contains(actual.Error(), exp) {
+				found = true
+				break
+			}
+		}
+		if !found {
+			t.Errorf("expected %q to contain one of %v", actual, exps)
+		}
```
Comment (tsandall) [inline internal/wasm/sdk/test/e2e/external_test.go:217]: Nice! We should never hit the default branch until we introduce some new way for policies to error.

**Review by tsandall — APPROVED** [review]
LGTM

**Pull request merged**